pkg/features/kube_features.go

```diff
@@ -1076,21 +1076,21 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	ImageMaximumGCAge: {Default: true, PreRelease: featuregate.Beta},
 
-	InTreePluginAWSUnregister: {Default: false, PreRelease: featuregate.Alpha},
+	InTreePluginAWSUnregister: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.32
 
-	InTreePluginAzureDiskUnregister: {Default: false, PreRelease: featuregate.Alpha},
+	InTreePluginAzureDiskUnregister: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.32
 
-	InTreePluginAzureFileUnregister: {Default: false, PreRelease: featuregate.Alpha},
+	InTreePluginAzureFileUnregister: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.32
 
-	InTreePluginGCEUnregister: {Default: false, PreRelease: featuregate.Alpha},
+	InTreePluginGCEUnregister: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.32
 
-	InTreePluginOpenStackUnregister: {Default: false, PreRelease: featuregate.Alpha},
+	InTreePluginOpenStackUnregister: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.32
 
 	InTreePluginPortworxUnregister: {Default: false, PreRelease: featuregate.Alpha},
 
 	InTreePluginRBDUnregister: {Default: false, PreRelease: featuregate.Deprecated}, // deprecated in 1.28, remove in 1.31
 
-	InTreePluginvSphereUnregister: {Default: false, PreRelease: featuregate.Alpha},
+	InTreePluginvSphereUnregister: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.32
```

Comment:
We'll need to test this on vSphere before this change is merged.

Reply:
FYI: 

- CSIMigrationvSphere is GA in 1.27. https://github.com/kubernetes/kubernetes/pull/116610.
- the CSIMigrationvSphere feature-gate was removed in 1.29. https://github.com/kubernetes/kubernetes/pull/121291
- in-tree code was removed in 1.30. https://github.com/kubernetes/kubernetes/pull/122090.

So 1.30 is the last release that can handle version skew between managers and nodes. It's not related to this PR.

Reply:
https://github.com/kubernetes/kubernetes/blob/227c2e7c2b2c05a9c8b2885460e28e4da25cf558/pkg/controller/volume/attachdetach/util/util.go#L271-L332

Reply:
According to the [https://kubernetes.io/releases/version-skew-policy/](https://kubernetes.io/releases/version-skew-policy/), kube-apiserver is at 1.29 and kubelet is supported at 1.29, 1.28, 1.27, and 1.26.

So if kube-apiserver is at 1.30 and kubelet is supported at 1.30, 1.29, 1.28, and 1.27. the in-tree code is removed in 1.30, we should revert it. cc @xing-yang @jsafrane

Reply:
the CSIMigrationvSphere feature-gate was GA at 1.27, so csinode's annotation `storage.alpha.kubernetes.io/migrated-plugins` always contains the `kubernetes.io/vsphere-volume` item.

Reply:
In other words, operations for the in-tree vsphereVolume type are redirected to the csi.vsphere.vmware.com CSI driver.

Reply:
cc @divyenpatel